typing/primitive.ml

```diff
@@ -69,6 +69,14 @@ let simple ~name ~arity ~alloc =
    prim_native_repr_args = make_native_repr_args arity Same_as_ocaml_repr;
    prim_native_repr_res = Same_as_ocaml_repr}
 
+let full ~name ~arity ~alloc ~native_name ~native_repr_args ~native_repr_res =
```

Comment:
I made the type `description` private to enforce the following invariant: `t.arity = List.length t.prim_native_repr_args`. I think it's better to not trust the user to  provide the correct arity, we should just do a `List.length native_repr_args` here.

For the name, I would slightly prefer just `make`.